### PR TITLE
 feat(zone): add resource migration

### PR DESCRIPTION
**Pull request by mgirouard**
Adds support for migrating `zone` resources.

Related PR for migration integration tests: https://github.com/cloudflare/terraform-provider-cloudflare/pull/6468

### CODE REVIEW

File: integration/v4_to_v5/testdata/zone/expected/zone.tf [hunk 1]
```diff
@@ -0,0 +1,311 @@
+# Comprehensive Zone Migration Integration Test
+# Covers all Terraform patterns and zone-specific edge cases
```
Comment (vaishakdinesh) [inline integration/v4_to_v5/testdata/zone/expected/zone.tf:2]: Love the testdata!

**Review by vaishakdinesh — APPROVED** [review]
